src/UI/Component/MainControls/Factory.php

```diff
@@ -0,0 +1,271 @@
+<?php
+
+/* Copyright (c) 2018 Nils Haagen <nils.haagen@concepts-and-training.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Component\MainControls;
+
+use \ILIAS\UI\Component\Image\Image;
+
+/**
+ * This is what a factory for main controls looks like.
+ */
+interface Factory {
+	/**
+	 * ---
+	 * description:
+	 *   purpose: >
+	 *     The Metabar is a unique page section to accomodate elements that
+	 *     should permamently be in sight of the user.
+	 *     The Metabar shall, first of all, host Prompts, i.e. notifications
+	 *     from the System to the user, but may also accomodate components and
+	 *     links deemed important, like help or search. The content of the bar
+	 *     does not change when navigating the system, but may depend on a
+	 *     configuration.
+	 *
+	 *   composition: >
+	 *     The Metabar is rendered horizontally at the very top of the page. It
+	 *     is always visible and available (except in some specialized view modes
+	 *     like an exam mode) as a static screen element and is unaffected by
+	 *     scrolling.
+	 *
+	 *     The Metabar always features the logo on the left hand side, while
+	 *     further elements are placed on the right hand side. Currently, these
```

Comment:
It could be more flexible to have the logo not being a part of the metabar and add it to the page layout directly imo.